main.c

```diff
@@ -0,0 +1,200 @@
+                               2, 
+                               alt_2_arm_state,
+                               (uint16_t)(ADCC_GetSingleConversion(A2_DROGUE_PIN))*3.72,
+                               (uint16_t)(ADCC_GetSingleConversion(A2_MAIN_PIN))*3.72,
+                               &alt_2_arm_stat_msg);
+            txb_enqueue(&alt_2_arm_stat_msg);
+            
+            can_msg_t bat_1_v_msg;
+            build_analog_data_msg(millis(), 
+                                  SENSOR_ARM_BATT_1, 
+                                  (uint16_t)(ADCC_GetSingleConversion(BATTERY_1_PIN))*3.72,
+                                  &bat_1_v_msg);
+            txb_enqueue(&bat_1_v_msg);
+            
+            can_msg_t bat_2_v_msg;
+            build_analog_data_msg(millis(), 
+                                  SENSOR_ARM_BATT_2, 
+                                  (uint16_t)(ADCC_GetSingleConversion(BATTERY_2_PIN))*3.72,
+                                  &bat_2_v_msg);
+            txb_enqueue(&bat_2_v_msg);
+            
+        }
+        
+        parse_altitude();
+        if (new_altitude_available()){
+            can_msg_t altitude_msg;
+            build_altitude_data_msg(millis(), get_altitude(), &altitude_msg);
+            txb_enqueue(&altitude_msg);
+        }
+        // set io to arm state of altimeter 1
+        if(alt_1_arm_state == DISARMED) {
+            DISARM_A1();
+            RED_LED_OFF();
+        }
+        else {
+            ARM_A1();
+            RED_LED_ON();
+        }
+        
+        // set io to arm state of altimeter 2
+        if(alt_2_arm_state == DISARMED) {
+            DISARM_A2();
+            BLUE_LED_OFF();
+        }
+        else {
+            ARM_A2();
+            BLUE_LED_ON();
+        }
+        
+        // send queued messages
+        txb_heartbeat();
+    }
+    return (EXIT_SUCCESS);
+}
+
+static void __interrupt() interrupt_handler(){
+    if (PIE3bits.TMR0IE == 1 && PIR3bits.TMR0IF == 1) {
+        timer0_handle_interrupt();
+        PIR3bits.TMR0IF = 0;
+    }
+    if (PIR5) {
+        can_handle_interrupt();
+    }
+    if(U1ERRIRbits.FERIF == 1 || U1ERRIRbits.RXFOIF){   //should probably do something if there is an error
+        RED_LED_ON();
+    }
+    else if (PIE3bits.U1RXIE == 1 && PIR3bits.U1RXIF == 1){
+        uart1_handle_interupt();
+        PIR3bits.U1RXIF = 0;
+    }
+    else if(PIR3bits.U1EIF == 1){   //should probably do something if there is an error
+        PIR3bits.U1EIF = 0;
+    }
+    
+}
+
+static void can_msg_handler(const can_msg_t *msg) {
+    uint16_t msg_type = get_message_type(msg);
+     
+    // ignore messages that were sent from this board
+    if (get_board_unique_id(msg) == BOARD_UNIQUE_ID) {
+        return;
+    }
+    
+    // declare this in advance cause we can't declare it inside the switch
+    uint8_t alt_num = 0;
+    enum ARM_STATE desired_arm_state = ARMED;
+    
+    switch (msg_type) {
+        case MSG_ALT_ARM_CMD:
```

Comment:
There's one more message you might care about: `MSG_GENERAL_CMD`. Radio board typically sends out a warning before the CAN bus is about to go down (usually because we turned off the RLCS box). I'm not sure if your board needs to do anything in this situation but it's worth noting.

Example from vent: https://github.com/waterloo-rocketry/cansw_vent/blob/020f12489f8bcc6523c28edb3b0a076923bd8aba/main.c#L170

Reply:
Arming also draws power from the recovery batteries so it shouldn't care.